tests/ci/cdk/util/fleet_config_loader.py

```diff
@@ -0,0 +1,200 @@
+            
+        Raises:
+            FileNotFoundError: If config file doesn't exist
+            yaml.YAMLError: If YAML parsing fails
+            ValueError: If configuration validation fails
+        """
+        config_path = pathlib.Path(config_path)
+        
+        if not config_path.exists():
+            raise FileNotFoundError(f"Configuration file not found: {config_path}")
+        
+        with open(config_path, 'r') as file:
+            raw_config = yaml.safe_load(file)
+        
+        return FleetConfigLoader._parse_config(raw_config)
+    
+    @staticmethod
+    def _parse_config(raw_config: Dict) -> FleetConfigurationFile:
+        """Parse raw YAML configuration into structured objects."""
+        # Parse global settings
+        global_settings_raw = raw_config.get('globalSettings', {})
+        global_settings = GlobalSettings(
+            default_tags=global_settings_raw.get('defaultTags', {})
+        )
+        
+        # Parse fleet configurations
+        fleets = []
+        for fleet_raw in raw_config.get('fleets', []):
+            fleet = FleetConfigLoader._parse_fleet(fleet_raw, global_settings)
+            fleets.append(fleet)
+        
+        return FleetConfigurationFile(
+            fleets=fleets,
+            global_settings=global_settings
+        )
+    
+    @staticmethod
+    def _parse_fleet(fleet_raw: Dict, global_settings: GlobalSettings) -> FleetConfiguration:
+        """Parse a single fleet configuration."""
+        # Parse scaling configuration
+        scaling_raw = fleet_raw.get('scalingConfiguration', {})
+        scaling_config = ScalingConfiguration(
+            enabled=scaling_raw.get('enabled', False),
+            utilization_rate=scaling_raw.get('utilizationRate'),
+            max_capacity=scaling_raw.get('maxCapacity', None),
+        )
+        
+        # Parse attribute-based compute if present
+        attribute_compute = None
+        if 'attributeBasedCompute' in fleet_raw:
+            attr_raw = fleet_raw['attributeBasedCompute']
+            attribute_compute = AttributeBasedCompute(
+                vcpus=attr_raw['vcpus'],
+                memory=attr_raw['memory'],
+                storage=attr_raw['storage']
+            )
+        
+        # Merge tags with global defaults
+        fleet_tags = fleet_raw.get('tags', {})
+        merged_tags = {**global_settings.default_tags, **fleet_tags}
+        
+        return FleetConfiguration(
+            name=fleet_raw['name'],
+            base_capacity=fleet_raw['baseCapacity'],
+            environment_type=fleet_raw['environmentType'],
+            scaling_configuration=scaling_config,
+            tags=merged_tags,
+            compute_type=fleet_raw.get('computeType'),
+            custom_instance_type=fleet_raw.get('customInstanceType'),
+            attribute_based_compute=attribute_compute
+        )
+    
+    @staticmethod
+    def get_default_config_path() -> pathlib.Path:
```

Comment:
What/where is error if path doesn't exist? This would probably be nice to surface as an error in the CI to debug more easily.